Build fails with imageStream for buildRecreate value

### Describe the bug

I ran into an issue where the plugin configuration element `buildRecreate` is not accepting the documented values of `imageStream` or `buildConfig`. Note that their respective shorthand versions `is` and `bc` are accepted fine and can be used as a workaround.

I have located the issue to be in [BuildRecreateMode::fromParameter](https://github.com/eclipse-jkube/jkube/blob/master/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/BuildRecreateMode.java). There is an implicit conversion to lowercase of the input value while the enum identifiers `imageStream` and `buildConfig` contain a capital letter, making the code fail.

### Eclipse JKube version

1.17.0

### Component

OpenShift Maven Plugin

### Apache Maven version

other (please specify in additional context)

### Gradle version

None

### Steps to reproduce

Any fragment like this:
```
               <configuration>
                   <registry>default-route-openshift-image-registry.apps-crc.testing/otcg/otcg-server</registry>
                    <buildStrategy>jib</buildStrategy>
                    <buildRecreate>imageStream</buildRecreate> <!
...
```

### Expected behavior

Expected value `imageStream` to be accepted for element `buildRecreate` as per documentation [here](https://eclipse.dev/jkube/docs/openshift-maven-plugin/#build-goal-configuration):

> buildRecreate
> 	
> If the build is performed in an OpenShift cluster then this option decides how the OpenShift resource objects associated with the build should be treated when they already exist:
> 
>     buildConfig or bc : Only the BuildConfig is recreated
> 
>     imageStream or is : Only the ImageStream is recreated
> 
>     all : Both, BuildConfig and ImageStream are recreated
> 
>     none : Neither BuildConfig nor ImageStream is recreated
> 
> The default is none. If you provide the property without value then all is assumed, so everything gets recreated.

### Runtime

OpenShift

### Kubernetes API Server version

other (please specify in additional context)

### Environment

Linux

### Eclipse JKube Logs

```shell
[ERROR] Failed to execute goal org.eclipse.jkube:openshift-maven-plugin:1.17.0:build (default-cli) on project otcg-websocket: Execution default-cli of goal org.eclipse.jkube:openshift-maven-plugin:1.17.0:build failed: No enum constant org.eclipse.jkube.kit.common.BuildRecreateMode.imagestream -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.eclipse.jkube:openshift-maven-plugin:1.17.0:build (default-cli) on project otcg-websocket: Execution default-cli of goal org.eclipse.jkube:openshift-maven-plugin:1.17.0:build failed: No enum constant org.eclipse.jkube.kit.common.BuildRecreateMode.imagestream
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:333)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.eclipse.jkube:openshift-maven-plugin:1.17.0:build failed: No enum constant org.eclipse.jkube.kit.common.BuildRecreateMode.imagestream
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:133)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
Caused by: java.lang.IllegalArgumentException: No enum constant org.eclipse.jkube.kit.common.BuildRecreateMode.imagestream
    at java.lang.Enum.valueOf (Enum.java:293)
    at org.eclipse.jkube.kit.common.BuildRecreateMode.valueOf (BuildRecreateMode.java:37)
    at org.eclipse.jkube.kit.common.BuildRecreateMode.fromParameter (BuildRecreateMode.java:60)
    at org.eclipse.jkube.maven.plugin.mojo.build.OpenshiftBuildMojo.generatorContextBuilder (OpenshiftBuildMojo.java:101)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.doExecute (AbstractDockerMojo.java:405)
    at org.eclipse.jkube.maven.plugin.mojo.build.OpenshiftBuildMojo.doExecute (OpenshiftBuildMojo.java:88)
    at org.eclipse.jkube.maven.plugin.mojo.build.AbstractDockerMojo.execute (AbstractDockerMojo.java:371)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:126)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute2 (MojoExecutor.java:328)
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:316)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:212)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:174)
    at org.apache.maven.lifecycle.internal.MojoExecutor.access$000 (MojoExecutor.java:75)
    at org.apache.maven.lifecycle.internal.MojoExecutor$1.run (MojoExecutor.java:162)
    at org.apache.maven.plugin.DefaultMojosExecutionStrategy.execute (DefaultMojosExecutionStrategy.java:39)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:159)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:105)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:73)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:53)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:118)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:261)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:173)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:101)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:906)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:283)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:206)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:103)
    at java.lang.reflect.Method.invoke (Method.java:580)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:255)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:201)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:361)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:314)
```


### Sample Reproducer Project

_No response_

### Additional context

Maven 3.9.9
CRC 4.16.7

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -23,6 +23,7 @@ Usage:
 ### 1.18-SNAPSHOT
 * Fix #1125: Support WebFlux SpringBoot projects when it comes to generate probes for actuators
 * Fix #2844: `oc:build` on openshift use `pods/log` to retrieve logs from build
+* Fix #3354: Build fails with `imageStream` for `buildRecreate` value
 
 ### 1.17.0 (2024-08-13)
 * Fix #494: Support for Micronaut Framework Native Images

--- a/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/BuildRecreateMode.java
+++ b/jkube-kit/common/src/main/java/org/eclipse/jkube/kit/common/BuildRecreateMode.java
@@ -57,7 +57,7 @@ public enum BuildRecreateMode {
         } else if (param.equalsIgnoreCase("true")) {
             return all;
         }
-        return valueOf(param.toLowerCase());
+        return valueOf(param);
     }
 
     BuildRecreateMode(boolean bc, boolean is) {

--- a/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/BuildRecreateModeTest.java
+++ b/jkube-kit/common/src/test/java/org/eclipse/jkube/kit/common/BuildRecreateModeTest.java
@@ -1,0 +1,42 @@
+/*
+ * Copyright (c) 2019 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at:
+ *
+ *     https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.jkube.kit.common;
+
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+class BuildRecreateModeTest {
+  static Stream<Arguments> fromParameterSource() {
+    return Stream.of(
+      arguments("buildConfig", BuildRecreateMode.buildConfig),
+      arguments("bc", BuildRecreateMode.bc),
+      arguments("imageStream", BuildRecreateMode.imageStream),
+      arguments("is", BuildRecreateMode.is),
+      arguments("all", BuildRecreateMode.all),
+      arguments("none", BuildRecreateMode.none)
+    );
+  }
+
+  @ParameterizedTest(name = "fromParameter({0}) should return {1}")
+  @MethodSource("fromParameterSource")
+  void fromParameter_shouldParseEnumCorrectly(String param, BuildRecreateMode expectedValue) {
+    assertThat(BuildRecreateMode.fromParameter(param)).isEqualTo(expectedValue);
+  }
+}

